tar.go

```diff
@@ -0,0 +1,154 @@
+package archiver
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func init() {
+	RegisterFormat("Tar", t{})
+}
+
+type t struct{}
+
+func (t) Match(filename string) bool {
+	// TODO: read file header to identify the format
```

Comment:
Note that when creating archives, the file won't exist yet, so we'd still have to fall back to the filename. (Not really relevant for this PR but just thought I'd mention it.)